rpc/src/helpers.rs

```diff
@@ -104,22 +105,89 @@ impl<C> PagedResult<C>
     }
 }
 
-#[derive(Serialize, Debug, Clone, Default)]
+// Enum defining possible response structures for RPC calls
+#[derive(Serialize, Debug, Clone)]
+pub enum RpcResponseData {
```

Comment:
Is there a reason why `Result<Vec<EndorsingRight>, RpcErrorMsg>` was not used here?

Reply:
Ocaml node has specific error message format as response to wrong RPC request or when specific error occurs during request, i.e.:
[{"kind":"permanent","id":"proto.005-PsBabyM1.context.storage_error","missing_key":["cycle","4","random_seed"],"function":"get"}]
[{"kind":"permanent","id":"proto.005-PsBabyM1.seed.unknown_seed","oldest":9,"requested":20,"latest":15}]

1. @jurajselep  fist question is if we want our errors to be same as ocaml node, this is not specified in Tezos RPC documentation (at least I did not found it) and maybe we should go for own errors.
2. one or other way it can be implemented also by adding context to Error and build response msg directly in result_to_json_response function

Reply:
there is description of this in code comment so I am resolving it here